packages/client/src/forms/register/mappings/query/field-mappings.ts

```diff
@@ -847,49 +828,50 @@ const isLocationFacilityOrCRVSOffice = (
   )
 }
 
-const transformAddressTemplateArray = (
+enum FHIRPropLocationLevel {
+  city,
+  district,
+  state,
+  country,
+  postalCode
+}
+
+const setAddressPropFromFHIRProp = (
   transformedData: IFormData,
   addressFromQuery: Address,
-  addressLocationLevel: keyof typeof LocationLevel,
+  fhirProp: keyof typeof FHIRPropLocationLevel,
   sectionId: string,
-  nameKey: string,
-  offlineData?: IOfflineData,
-  addressCases?: AddressCases
+  fieldName: string,
+  offlineData?: IOfflineData
 ) => {
-  if (!transformedData[sectionId]) {
-    transformedData[sectionId] = {}
-  }
-  if (!transformedData[sectionId][nameKey]) {
-    transformedData[sectionId][nameKey] = Array(3).fill('')
-  }
-  const addressName =
-    addressLocationLevel === 'country'
-      ? countries.find(
-          ({ value }) => value === addressFromQuery?.[addressLocationLevel]
-        )?.label || ''
+  const value =
+    fhirProp === 'country'
+      ? (countries.find(({ value }) => value === addressFromQuery?.[fhirProp])
+          ?.label.defaultMessage as string) || ''
       : offlineData?.[OFFLINE_LOCATIONS_KEY]?.[
-          addressFromQuery[addressLocationLevel] as string
+          addressFromQuery[
+            fhirProp as keyof typeof FHIRPropLocationLevel
+          ] as string
         ]?.name ||
-        addressFromQuery[addressLocationLevel] ||
+        addressFromQuery[fhirProp as keyof typeof FHIRPropLocationLevel] ||
         ''
-  ;(transformedData[sectionId][nameKey] as Array<string | MessageDescriptor>)[
-    LocationLevel[addressLocationLevel]
-  ] = addressName
-
-  const addressCase =
-    addressCases === AddressCases.SECONDARY_ADDRESS ? 'secondary' : 'primary'
-  transformedData[sectionId][
-    camelCase(`${nameKey}_${addressCase}_${addressLocationLevel}`)
-  ] = addressName as string | Record<string, string>
+  if (fhirProp === 'country') {
+    transformedData[sectionId][fieldName] = value
+  } else if (
+    addressFromQuery?.['country'] === window.config.COUNTRY &&
+    !fieldName.includes('international')
+  ) {
+    transformedData[sectionId][fieldName] = value
+  } else if (
+    addressFromQuery?.['country'] !== window.config.COUNTRY &&
+    fieldName.includes('international')
+  ) {
+    transformedData[sectionId][fieldName] = value
+  }
```

Comment:
These 2 cases are doing the same thing?

Reply:
I think its OK, because this function is called over and over for different fields and this stops duplication.  Consider that the "country" field comes in as Farajaland, and both "state" and "internationalState" fields look in FHIR for the "state" prop, so this method ends up setting both state and internationalState to the same thing, then you end up with a duplicate when the handlebars are side by side in the cert.  Consider the other case, country field comes in as Bangladesh, the same issue occurs.  So while the conditionals look the same, consider how the looping operates and the fact that in the cert you would have handlebars side by side like this: ```{{districtPlaceofmarriage}}{{internationalDistrictPlaceofmarriage}}, {{statePlaceofmarriage}}{{internationalStatePlaceofmarriage}}, {{countryPlaceofmarriage}}```

Reply:
The cert is configured for every possible case.